src/jit/optcse.cpp

```diff
@@ -880,39 +972,103 @@ void Compiler::optValnumCSE_InitDataFlow()
  */
 class CSE_DataFlow
 {
-    BitVecTraits* m_pBitVecTraits;
-    EXPSET_TP     m_preMergeOut;
+    Compiler* m_comp;
+    EXPSET_TP m_preMergeOut;
+    EXPSET_TP m_callKills;
 
 public:
-    CSE_DataFlow(Compiler* pCompiler) : m_pBitVecTraits(pCompiler->cseTraits), m_preMergeOut(BitVecOps::UninitVal())
+    CSE_DataFlow(Compiler* pCompiler)
+        : m_comp(pCompiler), m_preMergeOut(BitVecOps::UninitVal()), m_callKills(BitVecOps::UninitVal())
     {
+        m_callKills = BitVecOps::MakeEmpty(m_comp->cseLivenessTraits);
+        for (unsigned inx = 0; inx < m_comp->optCSECandidateCount; inx++)
```

Comment:
It seems that you should be able to factor out this code so that it isn't duplicated below.